src/components/SharedComponents/BottomSheet.js

```diff
@@ -1,53 +1,53 @@
 // @flow
 
-import {
-  BottomSheetBackdrop,
-  BottomSheetModal,
-  BottomSheetModalProvider
-} from "@gorhom/bottom-sheet";
+import BottomSheet, { BottomSheetView } from "@gorhom/bottom-sheet";
 import type { Node } from "react";
-import React, { useEffect, useRef } from "react";
+import React, {
+  useCallback, useEffect, useMemo, useRef
+} from "react";
 
 import { viewStyles } from "../../styles/sharedComponents/bottomSheet";
 
 type Props = {
-  children: any
+  children: any,
+  hasScrolled: boolean
 }
 
-const BottomSheet = ( { children }: Props ): Node => {
-  // ref
-  const bottomSheetModalRef = useRef( null );
-
-  const renderBackdrop = props => (
-    <BottomSheetBackdrop
-      // eslint-disable-next-line react/jsx-props-no-spreading
-      {...props}
-      pressBehavior="close"
-      appearsOnIndex={0}
-      disappearsOnIndex={-1}
-    />
-  );
+const StandardBottomSheet = ( { children, hasScrolled }: Props ): Node => {
+  const sheetRef = useRef( null );
+  const snapPoints = useMemo( () => ["45%"], [] );
 
-  useEffect( ( ) => {
-    // opens bottom sheet modal once when component first loads
-    bottomSheetModalRef.current?.present( );
+  // eslint-disable-next-line
+  const noHandle = ( ) => <></>;
+
+  const handleClosePress = useCallback( () => {
+    sheetRef.current?.close();
   }, [] );
 
+  const handleSnapPress = useCallback( ( ) => {
+    sheetRef.current?.snapToIndex( 0 );
+  }, [] );
+
+  useEffect( ( ) => {
+    if ( hasScrolled ) {
+      handleClosePress( );
+    } else {
+      handleSnapPress( );
+    }
+  }, [hasScrolled, handleClosePress, handleSnapPress] );
+
```

Comment:
IMO, this behavior should be optional, e.g. `<BottomSheet hideOnScroll />`. For example, Explore has a Bottom Sheet that might hide when you scroll in grid view, but not in map view.

Reply:
Ok, now it's optional, but as far as this component is concerned it has nothing to do with scrolling, right? It's just hiding the bottom sheet at the request of the parent component. Maybe the prop should just be `hide`?